fqd_http.c

```diff
@@ -262,6 +332,39 @@ fqd_http_message_headers_complete(http_parser *p) {
   }
   return 0;
 }
+
+#define cwrite(client, str) write(client->fd, str, strlen(str))
+
+static void
+fqd_http_jsend(remote_client *client, const char *status, const char *fmt, ...)
+{
+  char error[1024] = {0};
+  char scratch[1024] = {0};
+  const char *headers = "HTTP/1.0 200 OK\r\nConnection: close\r\nContent-Type: application/json\r\n\r\n";
+  va_list argp;
+
+  va_start(argp, fmt);
+  vsnprintf(error, sizeof(error), fmt, argp);
+  va_end(argp);
+
+  while(write(client->fd, headers, strlen(headers)) == -1 && errno == EINTR);
+  cwrite(client, "{\n");
+  cwrite(client, scratch);
+  cwrite(client,  " \"status\": \"");
+  cwrite(client, status);
+  cwrite(client,  "\",\n");
+  sprintf(scratch, " \"message\": \"%s\"\n", error);
+  cwrite(client, scratch);
+  cwrite(client, "}\n");
+}
+
+#define fqd_http_error_json_f(client, fmt, ...) fqd_http_jsend(client, "error", fmt, __VA_ARGS__)
+#define fqd_http_success_json_f(client, fmt, ...) fqd_http_jsend(client, "success", fmt, __VA_ARGS__)
+
+#define fqd_http_error_json(client, fmt) fqd_http_error_json_f(client, fmt, NULL)
+#define fqd_http_success_json(client, fmt) fqd_http_success_json_f(client, fmt, NULL)
```

Comment:
I think you want (client, "%s", fmt) otherwise you could crash by passing "%s%s%s" to this.